### DISCUSSION TITLE
Support for Continuous Query Notifications (CQN) feasible or already existing?

### DISCUSSION THREAD

**Discussion by MatthiasPetermann** [sijms/go-ora#294] [General]
Hello,

first of all a big respect and thank you that someone here has done the work to write an Oracle driver in pure Go. As a NetBSD user with the occasional wish to be able to access Oracle from this direction as well, this is a wonderful thing :-) At the moment there is still one feature keeping me from trying. And this is about CQN[1].

For those who may not have used it yet: CQN is, in short, a mechanism in Oracle that allows to register SQL queries on the database side (Subscribe), which are executed in the database whenever data changes. If the result set of these queries changes, the database sends an event over the connection to the client (publish). In terms of function, this is comparable to the Alert package, but much more powerful.

Long story short: what about the support in go-ora? Is this already available or planned? Or is it feasible at all?

Kind regards
Matthias

[1] https://docs.oracle.com/en/database/oracle/oracle-database/18/adfns/cqn.html

**Comment by sijms** [comment]
thank you 
I will read about this feature and try to add it in the driver
